src/signer_node/node_parameters.rs

```diff
@@ -74,14 +90,54 @@ impl<T: TapyrusApi> NodeParameters<T> {
             Ord::cmp(&a[..], &b[..])
         });
     }
+
+    /// Returns Map collection of received shares from all each signers in Key Generation Protocol
+    pub fn node_shared_secrets(&self) -> SharedSecretMap {
+        let mut secret_shares = SharedSecretMap::new();
+        for vss in &self.node_vss {
+            secret_shares.insert(
+                SignerID {
+                    pubkey: vss.sender_public_key,
+                },
+                SharedSecret {
+                    vss: VerifiableSS {
+                        parameters: ShamirSecretSharing {
+                            threshold: (self.threshold - 1) as usize,
+                            share_count: self.node_vss.len(),
+                        },
+                        commitments: vss
+                            .positive_commitments
+                            .iter()
+                            .map(|i| i.to_point())
+                            .collect(),
+                    },
+                    secret_share: vss.positive_secret,
+                },
+            );
+        }
+        secret_shares
+    }
+
+    /// Returns an aggregated share of the node.
+    pub fn node_secret_share(&self) -> SharedKeys {
+        let secret_shares = self.node_shared_secrets();
+
+        let shared_keys =
```

Comment:
Can you cache `shared_keys ` to memory in order to reduce the cost of re-calculation?

Reply:
I want to do that on federation module which is in PR #93. So i want to keep this part here.

Reply:
And, the PR #93 doesn't have any cache mechanism now. I want to do that after implementing serialization functionality because i'm not sure how we can treat struct members which is for on memory cache and should be skipped in serialization .